packages/eventual-send/src/index.d.ts

```diff
@@ -141,21 +161,29 @@ interface EProxy {
    * @param {*} x target for property get
    * @returns {ESingleGet} property get proxy
    */
-  readonly get: <T>(x: T) => ESingleGet<Unpromise<T>>;
+  readonly get: <T>(
+    x: T,
+  ) => ESingleGet<
+    T extends Remotable<infer U>
+      ? DataOnly<U>
+      : Awaited<T> extends Remotable<infer U>
+      ? DataOnly<U>
+      : Awaited<T>
+  >;
 
   /**
    * E.resolve(x) converts x to a handled promise. It is
    * shorthand for HandledPromise.resolve(x)
    */
-  readonly resolve: <T>(x: T) => Promise<Unpromise<T>>;
+  readonly resolve: <T>(x: T) => Promise<Awaited<T>>;
```

Comment:
Btw, what are our plans for `resolve` when we fold `HandledPromise` into `Promise`?

Reply:
> Btw, what are our plans for `resolve` when we fold `HandledPromise` into `Promise`?

Yeah, I've been thinking about that on and off.  Maybe we will have `Promise.delegated.resolve(x)` to return something that always waits one turn before resolving as a delegated promise?

Or just some encompassing behaviour for `Promise.resolve`?